article/test/PublicationControllerTest.scala

```diff
@@ -0,0 +1,135 @@
+  val bookSectionAgent = mock[NewspaperBookSectionTagAgent]
+  val publicationController = new PublicationController(bookAgent, bookSectionAgent)
+
+  "Publication Controller" should "redirect to an /all page when an observer dated book url is requested" in {
+    when(bookAgent.getTags("theobserver")).thenReturn(Seq(new TagDefinition("Observer Magazine","theobserver/magazine",None,false)))
+    when(bookSectionAgent.getTags("theobserver")).thenReturn(emptySeq)
+    val testReq = TestRequest("theobserver/2009/may/17/magazine")
+    val result = publicationController.publishedOn("theobserver","2009","may","17","magazine")(testReq)
+    status(result) should be(PermanentRedirect)
+    header("Location",result).getOrElse("") should endWith ("/theobserver/magazine/2009/may/17/all")
+  }
+
+  it should "redirect to an /all page when an observer dated book section url is requested" in {
+    when(bookAgent.getTags("theobserver")).thenReturn(emptySeq)
+    when(bookSectionAgent.getTags("theobserver")).thenReturn(Seq(new TagDefinition("News","theobserver/news/uknews",None,false)))
+    val testReq = TestRequest("theobserver/2015/nov/01/news/uknews")
+    val result = publicationController.publishedOn("theobserver", "2015", "nov", "01", "news/uknews")(testReq)
+    status(result) should be(PermanentRedirect)
+    header("Location",result).getOrElse("") should endWith ("/theobserver/news/uknews/2015/nov/01/all")
+  }
+
+  it should "200 when an observer article is requested" in {
+    when(bookAgent.getTags("theobserver")).thenReturn(emptySeq)
+    when(bookSectionAgent.getTags("theobserver")).thenReturn(emptySeq)
+    val testReq = TestRequest("theobserver/2014/aug/31/profile-clare-smyth-gordon-ramsay-chef-perfect-ten")
+    val result = publicationController.publishedOn("theobserver","2014","aug","31","profile-clare-smyth-gordon-ramsay-chef-perfect-ten")(testReq)
+    status(result) should be(OK)
+  }
+
+  it should "200 when an observer gallery is requested" in {
+    when(bookAgent.getTags("theobserver")).thenReturn(emptySeq)
+    when(bookSectionAgent.getTags("theobserver")).thenReturn(emptySeq)
+    val testReq = TestRequest("theobserver/gallery/2013/sep/14/the-10-best-fonts")
+    val result = publicationController.publishedOn("theobserver/gallery","2013","sep","14","the-10-best-fonts")(testReq)
+    status(result) should be(OK)
+  }
+
+//  it should "redirect to an /all page when an observer dated blog section is requested" in {
+//    when(bookAgent.getTags("theobserver/she-said")).thenReturn(emptySeq)
+//    when(bookSectionAgent.getTags("theobserver/she-said")).thenReturn(emptySeq)
+//    val testReq = TestRequest("theobserver/she-said/2015/jun/15")
+//    val result = controllers.PublicationController.publishedOn("theobserver/she-said","2015","jun","15","")(testReq)
+//    status(result) should be(PermanentRedirect)
+//    header("Location",result).getOrElse("") should endWith ("/theobserver/she-said/2015/jun/15/all")
+//  }
```

Comment:
This is one of those that won't test properly. They work when running (locally) so I'd like to make sure I don't forget this pattern when testing on code. Maybe I'll pop a reminder on the Trello card and ditch these from the test for now.